File: backend/components/metiers/administration_publique/utils/js/core/js_core.test.py
# js_core.test.py
# Tests unitaires Python pour js_core (équivalent JS, ultra avancé)
from .js_core import is_plain_object

def test_is_plain_object_dict():
    assert is_plain_object({'a': 1}) is True

def test_is_plain_object_list():
    assert is_plain_object([1,2,3]) is False

def test_is_plain_object_none():
    assert is_plain_object(None) is False

File: backend/components/metiers/administration_publique/utils/js/core/js_core.test.js
// js_core.test.js
// Tests unitaires JS pour js_core
const { isPlainObject } = require('./js_core');

describe('isPlainObject', () => {
  it('retourne true pour un objet pur', () => {
    expect(isPlainObject({ a: 1 })).toBe(true);
  });
  it('retourne false pour un tableau', () => {
    expect(isPlainObject([1,2,3])).toBe(false);
  });
  it('retourne false pour null', () => {
    expect(isPlainObject(null)).toBe(false);
  });
});
